pkg/controller/clustercontroller/reconcile.go

```diff
@@ -79,5 +79,7 @@ func (c *Controller) registerClusterInReconciliation(cluster *api.MysqlCluster)
 }
 
 func (c *Controller) removeClusterFromReconciliation(key string) {
-	c.clustersSync.Delete(key)
+	if ok1, ok2 := c.clustersSync.Load(key); ok2 && !ok1.(bool) {
```

Comment:
Sorry for confusion. I mean fixing line [257](https://github.com/presslabs/mysql-operator/blob/master/pkg/controller/clustercontroller/controller.go#L257) from [controller.go](https://github.com/presslabs/mysql-operator/blob/master/pkg/controller/clustercontroller/controller.go#L257), there is the real problem. And maybe renaming `ok2` -> `exists` and `ok1` -> `value`.

Here is not needed any check, `Delete` method ignores the key if it does not exists.